packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -310,6 +313,23 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         }
     }
 
+    public async tombstone() {
+        this.verifyNotClosed();
+        if (this.tombstoned) {
+            return;
+        }
+
+        assert(!(await this.isRoot()), "Tombstone should not be called on root datastores!");
```

Comment:
The data store should not be checking for root state. We want to move to a world where data stores don't know and don't care about this. If we do want to add this validation, it should be done by the callers. For example, GC could validate that root data stores are never marked unreferenced and thats it. It's all in one place and everything just works.

Reply:
I'd rather leave this here and keep this validation here as the logic for datastores sits on the context itself. When we finally move to a world where that knowledge sits elsewhere, we can remove it. What we can do is put a comment here indicating that this assert be removed when we move that logic out of the context. For now, it sits here.

Reply:
I will leave it up to you. But keep in mind that generally we should not add new dependencies on things we are planning to remove, unless necessary. Adding a comment doesn't help because its really hard to find them. You should create an issue to remove this and add it to the aliasing feature.

Couple other things to keep in mind:
1. Doing this makes the function async. This will make it hard to call this during container creation / loading.
2. This assert can hit on non-summarizer clients as well. Ideally, only the summarizer client should be checking this as it is responsible for generating tombstones.